### ISSUE TITLE
shon.xyz [ADS]

### ISSUE THREAD

**Issue by KiLL4FuNK** [uBlockOrigin/uAssets#2756]
URL(s) where the issue occurs

`http://shon.xyz/w1p5Y`

Describe the issue

ADS

![](https://k62.kn3.net/2/F/5/2/9/A/D7A.png)
Microsoft Windows 8.1 Version : Update 3 (64-bit)

![](https://k50.kn3.net/B/7/A/3/5/4/6AC.png)
Google Chrome Version : 67 (Official Build) (64-bit)

![](https://k50.kn3.net/D/D/4/5/E/9/CDB.png) 
uBlock Origin Version: 1.16.10

Settings

Default

**Comment by KiLL4FuNK** [comment]
Anti-Adblock

**Comment by KiLL4FuNK** [comment]
Anti-Adblock

**Comment by ghajini** [comment]
`https://shon.xyz/vPmlE`
3rd party tracking from different tracking ad servers at final download page
![IMG_20191203_114613](https://user-images.githubusercontent.com/20338483/70025293-9f611580-15c2-11ea-832c-75568b5fbcf7.jpg)

**Comment by ghajini** [comment]
I still see those 3rd party requests....

`https://user-images.githubusercontent.com/20338483/70036465-d7745280-15da-11ea-8a88-9213ffe313ed.jpg`

`Get link` button is unclickable
(after captcha completion )
clear cookies/data and change ip

**Comment by ghajini** [comment]
1.
OK they are smart, they avoid those 3rd party requests to same ip address again even if you clear cookies data.....

2. Now clear cookies data and change ip to other different from previous ip, they will serve ads and `get link` is unclickable again

**Comment by mapx-** [comment]
I blocked a remaining frame, testing FF desktop / android emulator
get link working fine for me

**Comment by ghajini** [comment]
Same problem  again

**Comment by okiehsch** [comment]
> Get link is unclickable on 1st visit, I have to reload the page to make it clickable.

Works on my end. I have to click twice on 1st visit but it works.

**Comment by mapx-** [comment]
mmhmm .. on (true) android the button is disabled.
@ghajini test these filters

```
||shon.xyz^$csp=script-src 'self' 'unsafe-inline' *.gstatic.com *.google.com *.googletagmanager.com *.googleapis.com;frame-src 'self' *.google.com,badfilter
||shon.xyz^$csp=script-src 'self' 'unsafe-inline' *.gstatic.com *.google.com *.googletagmanager.com *.googleapis.com;frame-src 'self' *.google.com;connect-src 'self'
shon.xyz#@#+js(window.open-defuser)
shon.xyz##+js(ra, onclick, [onclick^="window.open"])
```

**Comment by ghajini** [comment]
Filters at 
https://github.com/uBlockOrigin/uAssets/issues/2756#issuecomment-561114398

Results in popup ads at final download page

**Comment by mapx-** [comment]
I tested in android + FF, no popups on my end

**Comment by mapx-** [comment]
you could try
`shon.xyz##+js(aopr, _pop)`

**Comment by ghajini** [comment]
Sorry it doesn't help, I will try tomorrow

**Comment by ghajini** [comment]
Checked with opera browser also, there are popup ads at final page when clicking anywhere

**Comment by okiehsch** [comment]
I now also get popups.
We should restore 
`shon.xyz#@#+js(window.open-defuser)`
for all settings except Firefox android in my opinion.

**Comment by mapx-** [comment]
@ghajini @okiehsch 

could you test
`||shon.xyz^$csp=sandbox allow-forms allow-same-origin allow-scripts allow-modals allow-orientation-lock allow-pointer-lock allow-presentation allow-top-navigation`

**Comment by okiehsch** [comment]
That fixes the popup issue on my end using Chromium and Firefox non mobile.

**Comment by mapx-** [comment]
waiting @ghajini test directly on android

**Comment by mapx-** [comment]
I did test on android it seems ok on my end, the final button is working

**Comment by ghajini** [comment]
Fresh firefox installation, fresh ubo 1.12.4 installation, still popup ads on my side, this time firefox is preventing popups

`https://shon.xyz/LvWVU`

Screenshots=

**Comment by mapx-** [comment]
Do you see in the logger the last filter ?

**Comment by ghajini** [comment]
Yess I see

`https://user-images.githubusercontent.com/20338483/70147748-e9cdb980-16ca-11ea-83d1-5fee851f9f5b.jpg`

**Comment by mapx-** [comment]
In this case maybe firefox catched before uBo the popups (before uBo injecting its csp filter).
However, you have no popup / redirect right now.

**Comment by ghajini** [comment]
Yess right now firefox is preventing popups and not ubo, is upstream solution/fix available for this

**Comment by mapx-** [comment]
go into FF settings, disable the popup feature and see if uBo is blocking the popups

**Comment by ghajini** [comment]
> go into FF settings, disable the popup feature and see if uBo is blocking the popups

There is no such feature available on firefox android. .....

Forget this site, my fingers paining by repeated captcha solving and ip change in my vpn(I almost used all ip's😂😂😂)

**Comment by mapx-** [comment]
FF for android is allowing you to toggle the popup crap:

![image](https://user-images.githubusercontent.com/10303732/70150232-f2fa5e80-16a9-11ea-8b5d-91bfb0c82031.png)

**Comment by ghajini** [comment]
I toggled that feature to false, no popups on my end....but that is good feature by browser, it will prevent any unblocked popup not known to ublock origin....

Is there any tweak available for this site apart from disabling popup feature of browser

**Comment by mapx-** [comment]
it's already fixed, last filter prevents any popup for that page.

**Comment by mapx-** [comment]
read about sandbox here:
https://content-security-policy.com/

**Comment by ghajini** [comment]
Hello do you want to see my console(I enabled remote debugging )?

**Comment by mapx-** [comment]
Are you still talking about android ? opera ? still getting popups ?

**Comment by ghajini** [comment]
With popup feature default on firefox android, I get popups

With popup feature disabled on firefox android, I don't get popups

**Comment by ghajini** [comment]
Adguard browser extension for android works fluently ,why ublock origin lacking @gorhill

**Comment by gorhill** [comment]
> Fresh firefox installation, fresh ubo 1.12.4 installation

Latest stable uBO on Firefox is at 1.24.2.

**Comment by ghajini** [comment]
Yes iam using that only 1.24.2, it was typo 
![IMG_20191205_012207](https://user-images.githubusercontent.com/20338483/70176026-beb08d80-16fd-11ea-8848-75fea258c813.jpg)

**Comment by gorhill** [comment]
> I have reported to adguard, they have fixed it

Sure, but skimming through the numerous comment I can't figure what _exactly_ is the issue. I would need a well details but succinct steps to reproduce with no guessing on my part needed when I try to reproduce.

**Comment by gorhill** [comment]
Looking at Adguard fix, it's just a filter to neuter `window.open`, such scriptlet is also available in uBO.

**Comment by ghajini** [comment]
I will open new issue, if this issue tracker is misleading? Should I open?

**Comment by mapx-** [comment]
@gorhill the problem was with `window.open-defuser` => on the second page, only for android, the button `get link` was unclickable.

I removed `window.open-defuser` and added the `csp sandbox` filter which is working well on my end. I don't understand why ghajini still is getting popups (he is saying popups blocked by FF)

**Comment by ghajini** [comment]
With all default settings of ubo with uptodate filters and default firefox android settings,

If I browse 
`https://shon.xyz/LvWVU`
after captcha solving page, in next page when I click anywhere I get popup ads and  warning that firefox is preventing popups on this site and this no keeps on increasing, ubo blocking counter also increases exponentially....

Note -pls delete cookies/data and change ip if you have previously visited site with same ip

**Comment by ghajini** [comment]
`https://shon.xyz/2nTBi`
Still this appears at firefox android

![IMG_20200207_205714](https://user-images.githubusercontent.com/20338483/74042194-15339400-49ed-11ea-8f6c-1374847a0387.jpg)

**Comment by ghajini** [comment]
Can you add

`shon.xyz##+js(acis, decodeURI)`
`shon.xyz##+js(aopw, Fingerprint2)`
`shon.xyz##+js(acis, parseFloat)`
required for firefox android
